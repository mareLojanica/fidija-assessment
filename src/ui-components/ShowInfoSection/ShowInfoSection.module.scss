.content {
  display: grid;
  grid-template-columns: 1fr;
  gap: 10px;
  margin-top: 2rem;

  @media (max-width: 768px) {
    grid-template-columns: repeat(2, 1fr);
  }

  &__item {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 8px 0;
    border-bottom: 1px solid #ddd;
    @media (max-width: 768px) {
      border-bottom: none;
    }
  }

  &__label {
    flex: 3;
    font-weight: bold;
    color: #333;
  }

  &__value {
    flex: 2;
    color: #666;
    text-align: left;
  }

  @media (max-width: 768px) {
    &__item {
      flex-direction: column;
      align-items: flex-start;
      padding: 12px 0;
    }

    &__value {
      text-align: left;
      margin-top: 4px;
    }
  }
}
