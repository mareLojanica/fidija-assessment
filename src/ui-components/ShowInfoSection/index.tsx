import React, { useMemo } from "react"
import styles from "./ShowInfoSection.module.scss"
import { ShowInfoSectionProps } from "../../types/ui-component.types"
import ShowSection from "../ShowSection"

const ShowInfoSection: React.FC<ShowInfoSectionProps> = ({
	network,
	schedule,
	status,
	genres,
}) => {
	const infoData = useMemo(
		() => [
			{ label: "Streamed on", value: network ?? "Unknown" },
			{
				label: "Schedule",
				value: schedule.length ? schedule.join(", ") : "Not Available",
			},
			{ label: "Status", value: status },
			{
				label: "Genres",
				value: genres.length ? genres.join(", ") : "N/A",
			},
		],
		[network, schedule, status, genres]
	)

	return (
		<ShowSection title="Show Info">
			<ol className={styles.content} data-testid="show-info-section">
				{infoData.map(({ label, value }) => (
					<li key={label} className={styles.content__item}>
						<strong className={styles.content__label}>
							{label}:
						</strong>
						<span className={styles.content__value}>{value}</span>
					</li>
				))}
			</ol>
		</ShowSection>
	)
}

export default ShowInfoSection
