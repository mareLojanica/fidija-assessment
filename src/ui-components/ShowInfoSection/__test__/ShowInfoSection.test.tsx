import React from "react";
import { render, screen } from "@testing-library/react";
import ShowInfoSection from "../";
import "@testing-library/jest-dom";

describe("ShowInfoSection Component", () => {
  const mockShowInfo = {
    network: "RTS",
    schedule: ["Ponedeljak", "Utorak", "Sreda", "Cetvrtak", "Petak"],
    status: "Daje se",
    genres: ["Najbolja Serija Ikada"],
  };
  test("renders all show information correctly", () => {
    render(<ShowInfoSection {...mockShowInfo} />);

    expect(screen.getByText("Streamed on:")).toBeInTheDocument();
    expect(screen.getByText("RTS")).toBeInTheDocument();

    expect(screen.getByText("Schedule:")).toBeInTheDocument();
    expect(
      screen.getByText("Ponedeljak, Utorak, Sreda, Cetvrtak, Petak")
    ).toBeInTheDocument();

    expect(screen.getByText("Status:")).toBeInTheDocument();
    expect(screen.getByText("Daje se")).toBeInTheDocument();

    expect(screen.getByText("Genres:")).toBeInTheDocument();
    expect(screen.getByText("Najbolja Serija Ikada")).toBeInTheDocument();
  });

  test("handles missing network information", () => {
    render(<ShowInfoSection {...mockShowInfo} network={undefined} />);

    expect(screen.getByText("Streamed on:")).toBeInTheDocument();
    expect(screen.getByText("Unknown")).toBeInTheDocument();
  });

  test("handles missing schedule", () => {
    render(<ShowInfoSection {...mockShowInfo} schedule={[]} />);

    expect(screen.getByText("Schedule:")).toBeInTheDocument();
    expect(screen.getByText("Not Available")).toBeInTheDocument();
  });

  test("handles missing genres", () => {
    render(<ShowInfoSection {...mockShowInfo} genres={[]} />);

    expect(screen.getByText("Genres:")).toBeInTheDocument();
    expect(screen.getByText("N/A")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<ShowInfoSection {...mockShowInfo} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
