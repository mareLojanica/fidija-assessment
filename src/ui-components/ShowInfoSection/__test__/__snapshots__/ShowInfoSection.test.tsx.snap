// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowInfoSection Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="show-info"
  >
    <h2
      class="show-info__title"
    >
      Show Info
    </h2>
    <div
      class="show-info__content"
    >
      <ol
        class="content"
        data-testid="show-info-section"
      >
        <li
          class="content__item"
        >
          <strong
            class="content__label"
          >
            Streamed on:
          </strong>
          <span
            class="content__value"
          >
            RTS
          </span>
        </li>
        <li
          class="content__item"
        >
          <strong
            class="content__label"
          >
            Schedule:
          </strong>
          <span
            class="content__value"
          >
            Ponedeljak, Utorak, Sreda, Cetvrtak, Petak
          </span>
        </li>
        <li
          class="content__item"
        >
          <strong
            class="content__label"
          >
            Status:
          </strong>
          <span
            class="content__value"
          >
            Daje se
          </span>
        </li>
        <li
          class="content__item"
        >
          <strong
            class="content__label"
          >
            Genres:
          </strong>
          <span
            class="content__value"
          >
            Najbolja Serija Ikada
          </span>
        </li>
      </ol>
    </div>
  </div>
</DocumentFragment>
`;
